src/main/kotlin/org/neo4j/graphql/handler/filter/OptimizedFilterHandler.kt

```diff
@@ -340,8 +359,16 @@ class OptimizedFilterHandler(val type: GraphQLFieldsContainer) {
                         FieldOperator.values()
                             .map { it to definedField.name + it.suffix }
                             .mapNotNull { (predicate, queryFieldName) ->
-                                queriedFields.remove(queryFieldName)?.let { (index, objectField) ->
-                                    Predicate(predicate, objectField, definedField, index)
+                                queriedFields[queryFieldName]?.let { (index, objectField) ->
+                                    if (predicate.requireParam xor (objectField.value !is NullValue)) {
+                                        // if we got a value but the predicate requires none
```

Comment:
can we turn the comment into code instead?

Reply:
I did this before the `xor` but the condition was even uglier than the xor. Do you really don't like the `xor` so much.
In my opinion this line is even worse:
```kotlin
if ((predicate.requireParam && objectField.value is NullValue) || (!predicate.requireParam && objectField.value !is NullValue))
```